### PR TITLE
Mali - DS-Unit-2-Classification-2.md

### PR DESCRIPTION
**Pull request by mohamad-ali-nasser**

### CODE REVIEW

**Comment by axrd** [comment]
# Module 01 - Making Datasets, Avoiding Leakage

*Overall:* You're off to a decent start with the Lending Club data! I can tell that you've gotten pretty good at leveraging the power of Pandas!
*Feedback*: Get in the habit of being explicit about what you're doing and why. If you can explain your reasoning to a non-technical reader, you're quality of communication accounts for virtually every stakeholder you'll engage with. Leverage comments and text cells! Beyond that, try and recreate everything from lecture and ASK MORE QUESTIONS IF YOU GET STUCK!

## Get the data 
- Glob FTW! For future reference, it can be easier to assign your path to a variable and then use that to work with multiple files inside the same notebook. For example:
path = '/Users/YourUserName/Documents/etc/etc/FILES!'
- Good idea looking for Nulls. But you could Transpose the ouput and normalize it to see the percentage of missing values per feature. That way, features that have a TON of missing data you can drop for now and then only determine their predictive power *after* creating an initial model. Prioritize!

## Choose number of observations
- I see that you split the data but it would be great to add a text cell or some commenting to explain how/why you decided to do so (code etiquette). 

## Save your Data for reuse
- 👍 

## Choose Features
- Same as above. Document your choices more explicitly with text cells / comments.

**Comment by axrd** [comment]
# SC Review 

*Overall:* Really solid progress this week. Also kudos for stepping up to answer questions from your peers inside and outside Office Hours! 👏 👍  
*Feedback*: N/A you're progressing brilliantly. Let me know if there's anything you're not sure of ahead of the SC tomorrow! 

## Making Datasets, Avoiding Leakage
- Glob is a superpower. Remember that tomorrow in case you need to combine multiple csvs.  
- Key point of Data Hygiene in predictive modeling: leakage can easily lead you astray so proactively prevent it. Removing features is one really effective way to go about it. Consider Time-Series!

## Start with Baselines
- Not an explicit learning objective of this Sprint but so important that it's worth reiterating 👍 

## Class Imbalances
- When confronting classification problems, accuracy may not be a good measure because it doesn't factor True Positive rate. But the threshold for this is _contextual_. You want to bear in mind the *consequences* of your choices when factoring this in. Par for the course re: Predictive Modeling & Decision Making.
- If you determine that you can't afford false positives, use ROC AUC as a more appropriate metric for evaluating your model. By plotting the true positive rate vs the false positive rate at different thresholds, you can better understand your model's predictive power. 

## Making & Explaining Decisions
- The whole point of Data Science (if you can reduce it down to just one) is to draw actionable insights. That means making decisions. In order to better calibrate that decision-making, we leverage Expected Values, and Shapley Value Plots. Review the lecture notebook for links to dive deeper into both of these topics if you wish!

## Gradient Descent
- How can you programaticaly minimize a value you care about? Gradient Descent refers to a method that iteratively discovers minima. The trick is understanding if they are local or global, for one. For another, computation is not free. The time and effort it takes your model to minimize it's errors is a consideration you must make. Calibrate the learning_rate as one knob/dial.
